### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by enoqueJonas**
# OOP school library: set up associations

## This PR was raised to:
- Run a program using the command line.
- Set up associations between classes and objects.

## What does this PR do:
- Create a class Classroom with the following:
  - @label instance variable, should be initialized in the constructor.
  - Setter and getter for @label (remember about attr_accessor).
- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
  - Create the has-many side (a classroom has many students).
  - Create the belongs-to side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class Book with the following:
  - @title and @author instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about attr_accessor).
- Create a class Rental with the following:
  - @date instance variable, should be initialized in the constructor.
  - Setter and getter for @date (remember about attr_accessor).
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
  - Create the has-many side of Book and Rental (a book has many rentals).
  - Create the belongs-to side of Rental and Book (a rental belongs to a book).
  - Create the has-many side of Person and Rental (a person has many rentals).
  - Create the belongs-to side of Rental and Person (a rental belongs to a person).
  - Modify the constructor of Rental so Book and Person are set in it.

## Improvemts to be made:
- Add a UI to the app

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -24,5 +25,11 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```
Comment (VICTOR-CHUKWU) [inline person.rb]: - [ ] the add_rental method should take two parameters(as it is in your Rental class), date and book so as to create a new Rental using self as the third parameter(person), kindly fix this
```ruby
 def add_rental(date, book)
    Rental.new(date, book, self)
  end
```

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @enoqueJonas  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (enoqueJonas) [reply person.rb]: Hello @VICTOR-CHUKWU. I have a doubt. The rental that I will receive as an argument when I can the *add_rental* method won't be an instantiated object of class Rental?
Like this:
```ruby
rental = Rental.new(date, book, person)

add_rental(rental)

```

If so, I would not have to have *date* and *book* as arguments in the add_rental method.
Please let me know if my approach is the right one or no.

Comment (VICTOR-CHUKWU) [reply person.rb]: yes you are creating a new instance of Rental, hope the zoom call helped :+1:

Comment (enoqueJonas) [reply person.rb]: Thank you for clarifying @VICTOR-CHUKWU.

**Review by OLIPLICHE — APPROVED** [review]
Hi @enoqueJonas  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Applied the changes requested👏 👏
- Linter is green (the last commit have a green check mark (✅)) 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**